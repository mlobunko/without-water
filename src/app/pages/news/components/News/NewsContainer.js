import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as actions from '../../actions';
import { getAll } from '../../selectors';
import NewsComponent from './NewsComponent';

import error from 'shared/error';
import fetching from 'shared/fetching';

export class NewsContainer extends React.Component {
  componentDidMount = () => {
    this.props.errorActionsSetNull();
    if (!this.props.news.length && !this.props.fetching) {
      this.props.actionsFetch();
    }
  };

  render() {
    return (
      <NewsComponent
        fetching={this.props.fetching}
        isError={this.props.errorState.isError}
        serverError={this.props.errorState.serverError}
      />
    );
  }
}

const mapStateToProps = state => ({
  fetching: fetching.selectors.getAll(state),
  errorState: error.selectors.getAll(state),
  news: getAll(state)
});

const mapDispatchToProps = {
  actionsFetch: actions.fetch,
  errorActionsSetNull: error.actions.setNull
};
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(NewsContainer);

NewsContainer.propTypes = {
  fetching: PropTypes.bool,
  errorState: PropTypes.object,
  news: PropTypes.array
};
