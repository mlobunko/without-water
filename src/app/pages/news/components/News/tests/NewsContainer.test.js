import React from 'react';
import { shallow } from 'enzyme';

import { NewsContainer } from '../NewsContainer';

const initialProps = {
  fetching: false,
  errorState: {
    isError: false,
    serverError: '',
    loginError: ''
  },
  news: [],
  actionsFetch: jest.fn(),
  errorActionsSetNull: jest.fn()
};

describe('handlers', () => {
  describe('componentDidMount', () => {
    describe('not fetch', () => {
      const props = {
        ...initialProps,
        fetching: true
      };
      beforeEach(() => {
        jest.resetAllMocks();
        shallow(<NewsContainer {...props} />);
      });

      it('call errors to null', () => {
        expect(props.errorActionsSetNull).toHaveBeenCalledTimes(1);
      });

      it('fetch is not called', () => {
        expect(props.actionsFetch).toHaveBeenCalledTimes(0);
      });
    });

    describe('fetch', () => {
      const props = {
        ...initialProps
      };
      beforeEach(() => {
        jest.resetAllMocks();
        shallow(<NewsContainer {...props} />);
      });

      it('call errors to null', () => {
        expect(props.errorActionsSetNull).toHaveBeenCalledTimes(1);
      });

      it('call fetch', () => {
        expect(props.actionsFetch).toHaveBeenCalledTimes(1);
      });
    });
  });
});

describe('render', () => {
  const props = {
    ...initialProps
  };
  it('renders correctly', () => {
    const wrapper = shallow(<NewsContainer {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
