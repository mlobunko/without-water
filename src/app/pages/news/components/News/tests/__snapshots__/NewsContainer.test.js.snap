// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<NewsComponent
  fetching={false}
  isError={false}
  news={Array []}
  serverError=""
/>
`;
